src/collection/components/ShareCollectionModal.tsx

```diff
@@ -1,16 +1,140 @@
-import React, { FunctionComponent } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { get } from 'lodash-es';
+import React, { Fragment, FunctionComponent, useState } from 'react';
 
-import { Modal, ModalBody } from '@viaa/avo2-components';
+import {
+	Button,
+	FormGroup,
+	Modal,
+	ModalBody,
+	RadioButton,
+	RadioButtonGroup,
+	Spacer,
+	Toolbar,
+	ToolbarItem,
+	ToolbarRight,
+} from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import toastService, { TOAST_TYPE } from '../../shared/services/toast-service';
+import { UPDATE_COLLECTION_PROPERTY } from '../collection.gql';
 
 interface ShareCollectionModalProps {
 	isOpen: boolean;
 	setIsOpen: (isOpen: boolean) => void;
+	isPublic: boolean;
+	updateCollectionProperty: (value: boolean, fieldName: string) => void;
+	collection: Avo.Collection.Response;
 }
 
+const shareOptions = [
+	{
+		name: 'private',
+		label: 'Niet openbaar',
+		isPublicized: false,
+	},
+	{
+		name: 'public',
+		label: 'Openbaar',
+		isPublicized: true,
+	},
+];
+
 const ShareCollectionModal: FunctionComponent<ShareCollectionModalProps> = ({
 	setIsOpen,
 	isOpen,
+	isPublic,
+	collection,
+	updateCollectionProperty,
 }) => {
+	const [validationError, setValidationError] = useState();
+	const [isCollectionPublic, setIsCollectionPublic] = useState(isPublic);
+	const [triggerCollectionPropertyUpdate] = useMutation(UPDATE_COLLECTION_PROPERTY);
+
+	const validateFragments = (fragments: Avo.Collection.Fragment[]) => {
+		if (!fragments || !fragments.length) {
+			return false;
+		}
+
+		let isValid: Boolean = true;
+
+		// Check if fragment has custom_title and custom_description if necessary.
+		fragments.forEach(fragment => {
+			if (fragment.use_custom_fields && (!fragment.custom_title || !fragment.custom_description)) {
+				isValid = false;
+			}
+		});
+
+		return isValid;
+	};
+
+	const validateBeforeSave = () => {
+		const {
+			title,
+			description,
+			lom_classification,
+			lom_context,
+			collection_fragments,
+		} = collection;
+
+		// Validation ruleset
+		const validationObject = {
+			hasTitle: {
+				error: 'Uw collectie heeft geen titel.',
+				result: !!title,
+			},
+			hasDescription: {
+				error: 'Uw collectie heeft geen beschrijving.',
+				result: !!description,
+			},
+			hasContext: {
+				error: 'Uw collectie heeft geen vakken.',
+				result: !!(lom_context && lom_context.length),
+			},
+			hasClassification: {
+				error: "Uw collectie heeft geen onderwijsniveau's.",
+				result: !!(lom_classification && lom_classification.length),
+			},
+			hasAtleastOneFragment: {
+				error: 'Uw collectie heeft geen items.',
+				result: !!(collection_fragments && collection_fragments.length),
+			},
+			hasFullFragments: {
+				error: 'Uw items moeten een titel en beschrijving bevatten.',
+				result: validateFragments(collection_fragments),
```

Comment:
moeten deze echt full zijn? of is een titel OF een description voldoende?

Reply:
Moeten beide full zijn, heb het vanochtend nog eens nagevraagd omdat er in de feature-spreadsheet stond van niet, maar op dit ticket wel. Conclusie was dat beide filled in moeten zijn.